website/src/Menu.js

```diff
@@ -1,52 +1,131 @@
-import m from 'mithril'
-import './menu.css'
-import { User } from './User'
+import m, { render } from 'mithril'
 import { message } from 'alertifyjs'
+import { User } from './User'
+import LOGO from './assets/alWhite.png'
+import './styles/menu.scss'
 
-function Menu() {
+export function Menu() {
+  // toggles the navContainer visibility
   var open = false
   function toggle() {
     console.log('toggling menu', open)
     open = !open
   }
-  return {
-    view: (vnode) => {
-      return [
-        m('#menu.menu', {style: {display: open ? '': 'none', width: open ? '250px' : '0px'}}, [
-          m('i.material-icons', {onclick: toggle}, 'close'),
-        m(Links, {onclick: toggle}),
-        ]),
-        m('img.logo', {src: '/al-white.svg', onclick: toggle, style: {display: open ? 'none' : '' }}),
-      ]
-    }
-  }
-}
-function Link() {
-  return {
-    view: (vnode) => {
-      return m('.menu-item', [
-        m(m.route.Link, vnode.attrs, vnode.children)
-      ])
-    }
-  }
-}
 
-function Links() {
   return {
     view: vnode => {
       return [
-        m(Link, {href:'/', id: 'home-link', ...vnode.attrs}, 'home'),
-        m(Link, {href:'/projects', id: 'projects-link', ...vnode.attrs}, 'projects'),
-        m(Link, {href: '/renders', id: 'renders-link', ...vnode.attrs}, 'renders'),
-        m(Link, {
-          href: User.username ? '/logout' : '/login',
-          id: 'login-link',
-          ...vnode.attrs
-        }, User.username ? 'logout' : 'login'),
+        m(
+          'header',
+          m('.navbutton', m('i.material-icons', { onclick: toggle }, 'menu')),
+          m('.navContainer', { style: { width: open ? '250px' : '0px' } }, [
+            m(navLinks, { onclick: toggle }),
+          ]),
+          //TODO: change logo to new image in the future when location of images is known
+          m(
+            'div.logo',
+            m(
+              Link,
+              { href: '/', id: 'home-link', ...vnode.attrs },
+              m('img.logoImg', { src: LOGO })
+            )
+          ),
+          m('.menuLinks', m(Links)),
+          m(
+            Link,
+            {
+              class: 'loginLink',
+              href: User.username ? '/logout' : '/login',
+              id: 'login-link',
+              ...vnode.attrs,
+            },
+            User.username ? 'Logout' : 'Login'
+          )
+        ),
       ]
-    }
+    },
   }
 }
 
+//an individual link
+export const Link = () => ({
+  view: vnode => {
+    return m(
+      m.route.Link,
+      {
+        class: window.location.pathname.startsWith(vnode.attrs.href)
+          ? 'active'
+          : '',
```

Comment:
This replaces the `setActive` global helper function into a pure stateless function:

At render time, the component (in this case, a `Link`) checks if it's link destination is the start of the url path. If so, it sets the `active` class.

In Pseudoscript:
```javascript
if (pathname ~= this.href) {
  return {class: 'active'}
} else {
  return {class: ''}
}
```